apis/python/src/tiledbvcf/binding/reader.cc

```diff
@@ -225,6 +225,9 @@ void Reader::read(const bool release_buffs) {
   alloc_buffers(release_buffs);
   set_buffers();
 
+  // Release python GIL after we're done accessing python objects.
+  py::gil_scoped_release release;
```

Comment:
I would suggest putting this in its own inner scope block around only the expensive functions, to reduce the risk of eg accidentally adding some pybind11 object with function lifetime above in the future.

Reply:
Good idea, added the inner scope block.